### ISSUE TITLE
[macOS] Vulkan rendering on another thread causes pink border while resizing when using VK_PRESENT_MODE_FIFO_KHR

### ISSUE THREAD

**Issue by directmusic** [libsdl-org/SDL#7789]
Hello,

I am writing a multi-threaded renderer using Vulkan. In my application I am spawning a thread that handles the draw loop, but I handle all SDL Events on the main thread and use a message passing system to safely send them to the draw thread. I have been experiencing a pink border on macOS during a resize. At the moment I have not tested on Windows and Linux so it may be occurring there as well.

In my application, while using SDL2 I found that while using `VK_PRESENT_MODE_FIFO_KHR` to enable VSync I would see a pink border during resize in the bottom right corner however when switching to `VK_PRESENT_MODE_IMMEDIATE_KHR` the pink border would not show even if I was artificially limiting the FPS. My assumption is because the `SDL_MetalViewEventWatch()` callback in SDL_cocoametalview.m immediately processes the `SDL_WINDOWEVENT_SIZE_CHANGED` event and resizes the Metal layer before my renderer has finished creating a new frame at the last known size.

https://github.com/libsdl-org/SDL/blob/14f4fd3f11b4bc5bded6960c7c6118ea06ad86ac/src/video/cocoa/SDL_cocoametalview.m#L47-L53

My initial test in SDL2 was to comment out this line:

https://github.com/libsdl-org/SDL/blob/14f4fd3f11b4bc5bded6960c7c6118ea06ad86ac/src/video/cocoa/SDL_cocoametalview.m#L51

But found that it just caused me to never get a new size.

I switched to SDL3 and has the same pink border issue however now when commenting out that line to prevent it from updating the drawable size I no longer have the issue.

To ensure I wasn't just writing faulty Vulkan code I moved the Vulkan Cube demo code to an SDL Window. Here is a video demonstrating that.

https://github.com/libsdl-org/SDL/assets/26512622/19abc489-8dfd-4ff7-90e9-5098d42db23f

Here is the repo with the code: https://github.com/directmusic/sdl_multithreaded_window_resize_bug

I understand this may be out of SDL3's scope but if it is within scope I would be happy to spend time on the implementation and/or finding a more robust solution. Thanks.

**Comment by slouken** [comment]
@danginsburg, do you have any insight here?

**Comment by danginsburg** [comment]
> @danginsburg, do you have any insight here?

No, sorry, I would need to debug the issue.  The video here clearly shows it fixes the issue, but I'm not sure what impact there might be from removing that updateDrawableSize call.
